plugin/trino-hive/src/test/java/io/trino/plugin/hive/parquet/TestHiveParquetWithBloomFilters.java

```diff
@@ -0,0 +1,179 @@
+
+    private static final int NOT_EXIST_VALUE = 0;
+
+    @Test
+    public void testBloomFilterRowGroupPruning()
+            throws Exception
+    {
+        try (TempDataLocation tempDataLocation = new TempDataLocation("testFile", ".parquet", "testDir")) {
+            String tableName = "parquet_with_bloom_filters_" + randomNameSuffix();
+
+            createParquetBloomFilterSource(tempDataLocation.dataFile, COLUMN_NAME, TEST_VALUES);
+            assertUpdate(
+                    format(
+                            "CREATE TABLE %s" +
+                                    " (%s INT) " +
+                                    "WITH ( " +
+                                    "   format = 'PARQUET', " +
+                                    "   external_location = '%s' " +
+                                    ")",
+                            tableName, COLUMN_NAME, tempDataLocation.dataDirectory));
+
+            // when enabling reading bloomfilter, row groups are pruned when searching for a non-exist value
+            assertQueryStats(
+                    getSession(),
+                    String.format(String.format("SELECT * FROM %s WHERE %s IN (%s)", tableName, COLUMN_NAME, NOT_EXIST_VALUE)),
+                    queryStats -> {
+                        assertThat(queryStats.getPhysicalInputPositions()).isEqualTo(0);
+                        assertThat(queryStats.getProcessedInputPositions()).isEqualTo(0);
+                    },
+                    results -> assertThat(results.getRowCount()).isEqualTo(0));
+
+            // when enabling reading bloomfilter, row groups are not pruned when searching for an exist value
+            assertQueryStats(
+                    getSession(),
+                    String.format(String.format("SELECT * FROM %s WHERE %s IN (%s)", tableName, COLUMN_NAME, TEST_VALUES.get(0))),
+                    queryStats -> {
+                        assertThat(queryStats.getPhysicalInputPositions()).isGreaterThan(0);
+                        assertThat(queryStats.getProcessedInputPositions()).isEqualTo(queryStats.getPhysicalInputPositions());
+                    },
+                    results -> assertThat(results.getRowCount()).isEqualTo(1));
+
+            // when disabling reading bloomfilter, row groups are not pruned when searching for a non-exist value
+            assertQueryStats(
+                    disableBloomFilters(getSession()),
+                    String.format(String.format("SELECT * FROM %s WHERE %s IN (%s)", tableName, COLUMN_NAME, NOT_EXIST_VALUE)),
+                    queryStats -> {
+                        assertThat(queryStats.getPhysicalInputPositions()).isGreaterThan(0);
+                        assertThat(queryStats.getProcessedInputPositions()).isEqualTo(queryStats.getPhysicalInputPositions());
+                    },
+                    results -> assertThat(results.getRowCount()).isEqualTo(0));
+        }
+    }
+
+    @Override
```

Comment:
Place the `createQueryRunner()` method before any `@Test` method to be in sync with the other test classes from the project.